pkg/synchromanager/clustersynchro/cluster_monitor.go

```diff
@@ -61,11 +61,29 @@ func (synchro *ClusterSynchro) checkClusterHealthy() {
 	}
 
 	synchro.startResourceSynchro()
-	if lastReadyCondition.Status == metav1.ConditionTrue {
-		// TODO: if lastReadyCondition.Message != "", need process
+	message := "cluster health responded with ok"
+	if lastReadyCondition.Status == metav1.ConditionTrue && lastReadyCondition.Message == message {
 		return
 	}
 
+	if _, err := synchro.dynamicDiscoveryManager.GetAndFetchServerVersion(); err != nil {
+		message = fmt.Sprintf("cluster health responded with ok, but get server version: %v", err)
+	}
+
+	if lastReadyCondition.Status == metav1.ConditionTrue && lastReadyCondition.Message == message {
+		// the ready status and message has not been modified,
+		// to reduce the cluster status updates, do not update the `synchro.healthyCondition`.
+		return
+	}
+
+	synchro.healthyCondition.Store(metav1.Condition{
+		Type:               clusterv1alpha2.ClusterHealthyCondition,
+		Status:             metav1.ConditionTrue,
+		Reason:             clusterv1alpha2.ClusterHealthyReason,
+		Message:            message,
+		LastTransitionTime: metav1.Now().Rfc3339Copy(),
+	})
+
 	condition := metav1.Condition{
```

Comment:
The following duplicate logic can be removed

Reply:
fixed it ! Thank you